src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_gssapi.h"
+#include <assert.h>
+#include <string.h>
+
+static void NetSecurity_HandleError(uint32_t majorStatus, gss_buffer_t *outBufferHandle, uint32_t *outBufferLength)
+{
+    if (GSS_ERROR(majorStatus) || *outBufferHandle == nullptr || (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<int>::max()))
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
+    }
+}
+
+extern "C" uint32_t NetSecurity_AcquireCredSpNego(uint32_t* minorStatus, gss_name_t desiredName, int32_t isInitiate, gss_cred_id_t* outputCredHandle)
+{
+    assert(isInitiate == 0 || isInitiate == 1);
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {1, GSS_SPNEGO_MECHANISM};
```

Comment:
For all your struct initializations, please use the designated initializer syntax so you do not rely on the struct internal layout, meaning:
```
struct foo { 
int a;
int b;
}

Foo f = { .a = 5, .b = 6 };
```